policy/modules/system/systemd.te

```diff
@@ -13,6 +13,7 @@ attribute systemd_private_tmp_type;
 
 attribute systemd_read_efivarfs_type;
 fs_read_efivarfs_files(systemd_read_efivarfs_type)
+read_files_pattern(systemd_read_efivarfs_type, init_var_run_t, init_var_run_t)
```

Comment:
I see this questionable, too. Is the only reason the `/run/systemd/efivars/SystemdOptions-.*` file? Should we rather label the directory with a private type?

The `init_var_run_t` is assigned to many objects in /run/systemd which do not have their own type. With this rule, the access to generic systemd data is allowed to 14 systemd domains. Is this desirable?

(Another problem is overloading the "efivarfs" attribute with another meaning, but this is not easy to solve.)

Anyway, if there is the need, use an interface and move it below to local policy rules section.

Reply:
I consider this a temporary workaround. The cache for efivars in /run will hopefully go away as soon as the kernel ratelimit is fixed to take multiple readers into account. `init_var_run_t` is publicly readable data, so I don't think it makes sense to add a new type and make things more complicated for something that might go away next month.

Reply:
I understand. However, with this broad access some existing bugzillas can be made resolved, so we are about to introduce a planned regression.

Reply:
Yeah, but it's all public data. Both the stuff that was already under this label, and the "new" efivars stuff. It's publicly readable in the fs under /run, but a lot of it is also available over dbus to all unprivileged users. I'm not sure why that would be a "planned regression" — at the point where systemd stops providing the efivars cache, this rule can be dropped again and we return to status quo ante.

Reply:
So is this a temporary solution or a legit request? Note there are also objects with a different type. I was talking about regression as it looked you wanted to revert it back later.
```
# ls -Zago /run/systemd/
total 0
drwxr-xr-x. 21 system_u:object_r:init_var_run_t:s0                    480 Nov 12 11:56 .
drwxr-xr-x. 27 system_u:object_r:var_run_t:s0                         780 Nov 12 11:56 ..
drwxr-xr-x.  2 system_u:object_r:systemd_passwd_var_run_t:s0           40 Nov 12 11:56 ask-password
srw-------.  1 system_u:object_r:init_var_run_t:s0                      0 Nov 12 11:56 coredump
drwxr-xr-x.  7 system_u:object_r:systemd_unit_file_t:s0               300 Nov 12 11:56 generator
drwxr-xr-x.  3 system_u:object_r:init_var_run_t:s0                     60 Nov 12 11:56 generator.early
drwxr-xr-x.  2 system_u:object_r:init_var_run_t:s0                     60 Nov 12 11:56 home
drwxr-xr-x.  3 system_u:object_r:init_var_run_t:s0                    160 Nov 12 11:56 inaccessible
drwxr-xr-x.  2 system_u:object_r:systemd_logind_inhibit_var_run_t:s0   80 Nov 12 11:56 inhibit
drwxr-xr-x.  3 system_u:object_r:syslogd_var_run_t:s0                 180 Nov 12 11:56 journal
drwxr-xr-x.  2 system_u:object_r:systemd_machined_var_run_t:s0         40 Nov 12 11:56 machines
drwxr-xr-x.  5 system_u:object_r:systemd_networkd_var_run_t:s0        100 Nov 12 11:56 netif
srwxrwxrwx.  1 system_u:object_r:init_var_run_t:s0                      0 Nov 12 11:56 notify
srwxrwxrwx.  1 system_u:object_r:init_var_run_t:s0                      0 Nov 12 11:56 private
drwxr-xr-x.  3 system_u:object_r:systemd_resolved_var_run_t:s0        100 Nov 12 11:58 resolve
drwxr-xr-x.  2 system_u:object_r:systemd_logind_var_run_t:s0           60 Nov 12 11:56 seats
drwxr-xr-x.  2 system_u:object_r:systemd_logind_sessions_t:s0          80 Nov 12 11:57 sessions
drwxr-xr-x.  2 system_u:object_r:systemd_logind_var_run_t:s0           40 Nov 12 11:56 shutdown
drwxr-xr-x.  2 system_u:object_r:systemd_unit_file_t:s0                40 Nov 12 11:56 system
drwxr-xr-x.  2 system_u:object_r:init_var_run_t:s0                     60 Nov 12 11:57 transient
drwx------.  2 system_u:object_r:init_var_run_t:s0                     40 Nov 12 11:56 unit-root
drwxr-xr-x.  2 system_u:object_r:systemd_unit_file_t:s0              1080 Nov 12 11:58 units
drwxr-xr-x.  2 system_u:object_r:systemd_userdbd_runtime_t:s0         120 Nov 12 11:56 userdb
drwxr-xr-x.  2 system_u:object_r:systemd_logind_var_run_t:s0           60 Nov 12 11:57 users
```

Reply:
> So is this a temporary solution or a legit request?

Both? It's temporary in the sense that it hopefully will not be necessary forever. But there are no plans that I know of to change this in the kernel, so it'll be necessary at least until current and upcoming kernels go out of scope, i.e. a long while.

> Note there are also objects with a different type.

I don't understand the significance of objects with a different type, since this rule doesn't matter for them at all. 

So... I think the current rule is OK because it a) solves the issue at hand, b) it does allow access to more stuff than strictly necessary, but that stuff is public anyway so it's OK to allow access to it.

Reply:
@keszybz thank you for your effort to prepare this complex set of patches. I will merge this PR now.

Note removing rules from policy is troublesome as it can have unpredictable effect, depending on changes in other components requirement in the near future.